.github/workflows/build.yml

```diff
@@ -217,7 +217,7 @@ jobs:
     name: SonarCloud
     runs-on: ubuntu-20.04
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
         with:
           fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of analysis
       - name: Check coverage report existence
```

Comment:
I think there is a breaking change around the code coverage report that'll need to be debugged a bit further:
![image](https://github.com/Sage-Bionetworks/synapsePythonClient/assets/17128019/53df4d39-eec9-491f-8af1-6809b33e05ed)

The coverage.xml does exist, but is packaged into a zip. I don't recall the exact specifics of how this was working before:
![image](https://github.com/Sage-Bionetworks/synapsePythonClient/assets/17128019/9a523d9b-75d2-4893-af42-68f73fa9e844)

I believe that this was broken as a result of some changes that @danlu1 had made here:
https://github.com/Sage-Bionetworks/synapsePythonClient/pull/1095

The code coverage graph in sonarcloud coincides with this timeline where `Covered Lines` goes to 0:
![image](https://github.com/Sage-Bionetworks/synapsePythonClient/assets/17128019/5c70400b-0bc4-4d70-8bb3-1c834905617c)


Would you want to resolve this issue here, or do you think another ticket is appropriate?

Reply:
@BryanFauble I'll make a separate Jira ticket and PR for this, for book-keeping purposes, but can definitely investigate it this week. Would you mind sharing where I can find the contents in the last two screenshots?

Reply:
@jaymedina 
The coverage-report is here: https://github.com/Sage-Bionetworks/synapsePythonClient/actions/runs/9620252513?pr=1112

The historical code coverage graph is available in sonarcloud: https://sonarcloud.io/project/overview?id=Sage-Bionetworks_synapsePythonClient

Reply:
Thanks! I'll be merging this then and starting this separately.